### DISCUSSION TITLE
icon-library 2.29.0 fails to compile using Typescript 3.7.2

### DISCUSSION THREAD

**Discussion by s24y85u2** [Royal-Navy/design-system#1911] [Q&A] [answered]
The line "export type { SVGIconProps } from './types'" that was added in the following commit: https://github.com/Royal-Navy/design-system/commit/9730adfa21b340deae258fac204d9962c53f6862 does not compile with Typescript version 3.7.2 that we are currently using. The error we get is:

"Declaration or statement expected" and "';' expected"

We are using Typescript 3.7.2 and icon-library 2.29.0.

**Comment by m7kvqbe1** [comment, marked as answer]
We recommend upgrading TypeScript to the next major (4) or at least bumping the minor. 

The feature causing your build to fail (type-only imports and exports) was added to TypeScript in 3.8 (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html).

We are using the latest version of TypeScript throughout the design system now.
